lib/pure/os.nim

```diff
@@ -1254,11 +1254,11 @@ proc findExe*(exe: string, followSymlinks: bool = true;
         when not defined(windows):
           while followSymlinks: # doubles as if here
             if x.symlinkExists:
-              var r = newString(maxSymlinkLen)
-              var len = readlink(x.cstring, r.cstring, maxSymlinkLen)
+              var r = newString(maxPathLen)
+              var len = readlink(x.cstring, r.cstring, maxPathLen)
               if len < 0:
                 raiseOSError(osLastError(), exe)
-              if len > maxSymlinkLen:
+              if len > maxPathLen:
```

Comment:
IIRC, for max component length (filename), macos's default FS is 255 utf-8 chars, while btrfs 255 bytes, etc... see wikipedia limits link in previous comment.

I think we should keep to conservative maximums as that might be easier or get fancy with conditional limits based on target platform.

Reply:
That's the whole path, not a component.

On Linux, the symlink content can be any string, not even a valid file name. It's a text file with a special flag.